File: packages/backend/src/models/job.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from enum import Enum

class JobStatus(str, Enum):
    ACTIVE = "active"
    CLOSED = "closed"
    APPLIED = "applied"

class JobSource(str, Enum):
    LINKEDIN = "linkedin"
    INDEED = "indeed"
    GLASSDOOR = "glassdoor"
    COMPANY_WEBSITE = "company_website"

class JobBase(BaseModel):
    title: str
    company: str
    location: Optional[str] = None
    description: str
    requirements: Optional[str] = None
    salary_min: Optional[int] = None
    salary_max: Optional[int] = None
    job_type: Optional[str] = None  # full-time, part-time, contract
    remote: bool = False
    url: str
    source: JobSource
    posted_date: Optional[datetime] = None

class JobCreate(JobBase):
    pass

class JobUpdate(BaseModel):
    title: Optional[str] = None
    company: Optional[str] = None
    location: Optional[str] = None
    description: Optional[str] = None
    requirements: Optional[str] = None
    salary_min: Optional[int] = None
    salary_max: Optional[int] = None
    job_type: Optional[str] = None
    remote: Optional[bool] = None
    status: Optional[JobStatus] = None

class Job(JobBase):
    id: int
    status: JobStatus = JobStatus.ACTIVE
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

File: packages/shared/src/types/job.ts
export interface Job {
  id: number;
  title: string;
  company: string;
  location?: string;
  description: string;
  requirements?: string;
  salary_min?: number;
  salary_max?: number;
  job_type?: string;
  remote: boolean;
  url: string;
  source: JobSource;
  status: JobStatus;
  posted_date?: Date;
  created_at: Date;
  updated_at: Date;
}

export interface JobCreate {
  title: string;
  company: string;
  location?: string;
  description: string;
  requirements?: string;
  salary_min?: number;
  salary_max?: number;
  job_type?: string;
  remote: boolean;
  url: string;
  source: JobSource;
  posted_date?: Date;
}

export enum JobStatus {
  ACTIVE = "active",
  CLOSED = "closed",
  APPLIED = "applied"
}

export enum JobSource {
  LINKEDIN = "linkedin",
  INDEED = "indeed",
  GLASSDOOR = "glassdoor",
  COMPANY_WEBSITE = "company_website"
}